ci/test-docker-in-docker.sh

```diff
@@ -23,8 +23,8 @@ ci_dir=$(realpath "${ci_dir}")
 . "${ci_dir}"/shared.sh
 
 main() {
-    docker run -v "${PROJECT_HOME}":"${PROJECT_HOME}" -w "${PROJECT_HOME}" \
-        --rm -e TARGET -e RUSTFLAGS -e RUST_TEST_THREADS \
+    docker run --platform linux/amd64 -v "${PROJECT_HOME}":"${PROJECT_HOME}" -w "${PROJECT_HOME}" \
```

Comment:
added this as a bonus, not really needed but I had to add it to run on my arm64

Reply:
I think this is great for consistency, since we might not be deploying ARM64 images for a while and until #982 is merged, only `linux/amd64` will work anyway.